src/Spectre.AngularClient/src/app/divik-results/shared/divik.service.ts

```diff
@@ -24,53 +24,52 @@ import 'rxjs/Rx';
+import { HeatmapUtil } from '../../heatmaps/shared/heatmap.service';
 
 @Injectable()
 export class DivikService extends Service {
 
-    constructor(private http: Http) { super(); }
+  constructor(private http: Http) {
+    super();
+  }
 
-    private getHeaders() {
-      const headers = new Headers();
-      headers.append('Accept', 'application/json');
-      return headers;
-    }
+  private getHeaders() {
+    const headers = new Headers();
+    headers.append('Accept', 'application/json');
+    return headers;
+  }
 
-    get(preparationId: number, divikId: number, level: number): Observable<Heatmap> {
-      const queryUrl = `${this.getBaseUrl()}/divikResult/${preparationId}?divikId=${divikId}&level=${level}`;
-      const response = this.http.get(queryUrl, {headers: this.getHeaders()});
-      return response.map(toHeatmap);
-    }
+  get (preparationId: number, divikId: number, level: number): Observable<Heatmap> {
+    const queryUrl = `${this.getBaseUrl()}/divikResult/${preparationId}?divikId=${divikId}&level=${level}`;
+    const response = this.http.get(queryUrl, {headers: this.getHeaders()});
+    return response.map((res: Response) => HeatmapUtil.toHeatmap(res, '[DivikService]'));
+  }
+
+  getConfig(preparationId: number, divikId: number): Observable<DivikConfig> {
+    const queryUrl = `${this.getBaseUrl()}/divikResult/${preparationId}?divikId=${divikId}`;
+    const response = this.http.get(queryUrl, {headers: this.getHeaders()});
+    return response.map(toDivikConfig);
+  }
 }
 
-function toHeatmap(response: Response): Heatmap {
-    console.log('[DivikService] parsing response:');
-    console.log(response);
-    const json = response.json();
-    console.log(json);
-    const max_column = Math.max.apply(null, json.X);
-    const min_column = Math.min.apply(null, json.X);
-    const max_row = Math.max.apply(null, json.Y);
-    const min_row = Math.min.apply(null, json.Y);
-    console.log('[DivikService] found bounds: (' + min_row + ', ' + max_row + '), (' + min_column + ', ' + max_column + ')');
-    console.log('[DivikService] initializing heatmap');
-    const data = [];
-    for (let i = min_row; i < max_row + 1; i++) {
-      data[i - min_row] = [];
-      for (let j = min_column; j < max_column + 1; j++) {
-        data[i - min_row][j - min_column] = 0;
-      }
-    }
-    console.log('[DivikService] processing heatmap');
-    for (let i = 0; i < json.X.length; i++) {
-        const column = json.X[i] - min_column;
-        const row = max_row - json.Y[i];
-        const divikData = json.Data[i];
-        console.log('[DivikService] processing (' + row + ', ' + column + ', ' + divikData + ')');
-        data[row][column] = divikData;
-    }
-    console.log('[DivikService] heatmap processed');
-    return <Heatmap>({
-        data: data
-    });
+function toDivikConfig(response: Response): DivikConfig {
+  const json = response.json();
+  const properties = {};
+  properties['Max K'] = json.MaxK;
```

Comment:
Can't these be fields + mapping defined somewhere? While in this case your version is shorter and uber-fancy, it would be much more clear in general case. We want to elaborate some patterns, which will allow us to simply understand the code, even if someone absolutely new comes and investigates it.

Reply:
I do this to pretty display config using for, I think it look better than making string from field. It isn't so important to separate in a lot of field and it can change but if you want I can do this

Reply:
I thought of specification of consistent format, which can be checked during packaging of the application. Now, we specify all of those entries by hand either way, so it should not differ that much. If we investigate however, how this interface works, right now it's impossible from its file.